logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -56,34 +63,51 @@ def receive(event)
   def worker_setup
     if @workers == 1
       @worker_plugins = [self]
-      @worker_threads = []
     else
-      define_singleton_method(:handle, method(:handle_worker))
-      @worker_queue = SizedQueue.new(20)
+      define_singleton_method(:multi_handle, method(:handle_worker))
```

Comment:
oh, I just got it. there is already a `multi_handle` defined in the class which is the one used by the actual workers and the `define_singleton_method(:multi_handle, method(:handle_worker))` redefinition is only for the *top* worker which will pass events downs to the downstream output workers. 

I wonder how we could either name or in some way organize the code or document or whatever that would make it more obvious for someone in the code to understand this mechanic. even knowing about how it works it is really hard to understand. I believe we should take the opportunity of the current refactor to try and make that better?